lib/tools.js

```diff
@@ -1,1577 +1,1615 @@
+ * @param {boolean} [isNonEdit] optional indicator if copy is in nonEdit part
+ *
+ */
+function copyAttributes(oldObj, newObj, originalObj, isNonEdit) {
+    for (const attr in oldObj) {
+        if (oldObj.hasOwnProperty(attr)) {
+            if (typeof oldObj[attr] !== 'object' || oldObj[attr] instanceof Array) {
+                if (oldObj[attr] === '__no_change__' && originalObj && !isNonEdit) {
+                    if (originalObj[attr] !== undefined) {
+                        newObj[attr] = JSON.parse(JSON.stringify(originalObj[attr]));
+                    } else {
+                        console.log(`Attribute ${attr} ignored by copying`);
+                    }
+                } else
+                if (oldObj[attr] === '__delete__' && !isNonEdit) {
+                    if (newObj[attr] !== undefined) {
+                        delete newObj[attr];
+                    }
+                } else {
+                    newObj[attr] = oldObj[attr];
+                }
+            } else {
+                newObj[attr] = newObj[attr] || {};
+                copyAttributes(oldObj[attr], newObj[attr], originalObj && originalObj[attr], isNonEdit || attr === 'nonEdit');
+            }
+        }
+    }
+}
+
+/**
+ * Checks the flag nonEdit and restores non-changeable values if required
+ *
+ * @alias checkNonEditable
+ * @memberof tools
+ * @param {object} oldObject source object
+ * @param {object} newObject destination object
+ *
+ */
+function checkNonEditable(oldObject, newObject) {
+    if (!oldObject) return true;
+    if (!oldObject.nonEdit && !newObject.nonEdit) return true;
+
+    // if nonEdit is protected with password
+    if (oldObject.nonEdit && oldObject.nonEdit.passHash) {
+        // If new Object wants to update the nonEdit information
+        if (newObject.nonEdit && newObject.nonEdit.password) {
+            crypto = crypto || require('crypto');
+            const hash = crypto.createHash('sha256').update(newObject.nonEdit.password.toString()).digest('base64');
+            if (oldObject.nonEdit.passHash !== hash) {
+                delete newObject.nonEdit;
+                return false;
+            } else {
+                oldObject.nonEdit = JSON.parse(JSON.stringify(newObject.nonEdit));
+                delete oldObject.nonEdit.password;
+                delete newObject.nonEdit.password;
+                oldObject.nonEdit.passHash = hash;
+                newObject.nonEdit.passHash = hash;
+            }
+            copyAttributes(newObject.nonEdit, newObject, newObject);
+
+            if (newObject.passHash) delete newObject.passHash;
+            if (newObject.nonEdit && newObject.nonEdit.password) delete newObject.nonEdit.password;
+
+            return true;
+        } else {
+            newObject.nonEdit = oldObject.nonEdit;
+        }
+
+    } else if (newObject.nonEdit) {
+        oldObject.nonEdit = JSON.parse(JSON.stringify(newObject.nonEdit));
+        if (newObject.nonEdit.pas
```

Comment:
```diff
- measureEventLoop
+ measureEventLoopLag
```
Those 3 letters make it pretty clear what we're measuring. No need to save 3 bytes ;)

Reply:
will change